### PR TITLE
feat: initialise special groups on bootstrap

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
groups / bootstrap 

### Description
This PR initialises special groups on bootstrap

### Steps

- [x] Added implementation
- [x] Added / updated tests
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

File: src/permissions/permissions.service.ts [hunk 1]
```diff
@@ -102,12 +103,12 @@ export class PermissionsService {
       if (groupPermission.canEdit || !wantEdit) {
         // Handle special groups
         if (groupPermission.group.special) {
-          if (groupPermission.group.name == 'loggedIn') {
+          if (groupPermission.group.name == SpecialGroup.LOGGED_IN) {
             // TODO: Name of group for logged in users
```
Comment (InnayTool) [inline src/permissions/permissions.service.ts:107]: I think this can be removed now.

**Review by InnayTool — CHANGES_REQUESTED** [review]

File: src/permissions/permissions.service.ts (hunk 1 above)
Comment (DerMolly) [reply src/permissions/permissions.service.ts]: Yeah, I guess so. Did that

**Review by InnayTool — APPROVED** [review]

File: src/groups/groups.service.spec.ts [hunk 2]
```diff
@@ -46,6 +46,32 @@ describe('GroupsService', () => {
...
+      jest
+        .spyOn(groupRepo, 'save')
+        .mockImplementationOnce(async (group: Group): Promise<Group> => group);
+    });
+    it('works', async () => {
+      const user = await service.createGroup(groupName, displayname);
+      expect(user.name).toEqual(groupName);
+      expect(user.displayName).toEqual(displayname);
+    });
+    it('fails if username is already taken', async () => {
```
Comment (ErikMichelson) [inline src/groups/groups.service.spec.ts:62]:
```suggestion
    it('fails if group name is already taken', async () => {
```

File: src/groups/groups.service.spec.ts [hunk 3]
```diff
@@ -46,6 +46,32 @@ describe('GroupsService', () => {
...
+      jest.spyOn(groupRepo, 'save').mockImplementationOnce(() => {
+        throw new Error();
+      });
+      // create first user with username
```
Comment (ErikMichelson) [inline src/groups/groups.service.spec.ts:66]:
```suggestion
      // create first group with group name
```

File: src/groups/groups.service.spec.ts [hunk 4]
```diff
@@ -46,6 +46,32 @@ describe('GroupsService', () => {
...
+      await service.createGroup(groupName, displayname);
+      // attempt to create second user with username
```
Comment (ErikMichelson) [inline src/groups/groups.service.spec.ts:68]:
```suggestion
      // attempt to create second group with group name
```

File: src/groups/groups.service.ts [hunk 5]
```diff
@@ -21,6 +21,35 @@ export class GroupsService {
     this.logger.setContext(GroupsService.name);
   }
 
+  /**
+   * @async
+   * Create a new group with a given name and displayName
+   * @param name - the username the new user shall have
```
Comment (ErikMichelson) [inline src/groups/groups.service.ts:27]:
```suggestion
   * @param name - the group name the new group shall have
```

File: src/groups/groups.service.ts [hunk 6]
```diff
@@ -21,6 +21,35 @@ export class GroupsService {
...
+   * @param displayName - the display the new user shall have
```
Comment (ErikMichelson) [inline src/groups/groups.service.ts:28]:
```suggestion
   * @param displayName - the display the new group shall have
```

File: src/groups/groups.service.ts [hunk 7]
```diff
@@ -21,6 +21,35 @@ export class GroupsService {
...
+    displayName: string,
+    special = false,
+  ): Promise<Group> {
+    const group = Group.create(name, displayName);
+    group.special = special;
+    try {
+      return await this.groupRepository.save(group);
+    } catch {
+      this.logger.debug(
+        `A group with the username '${name}' already exists.`,
```
Comment (ErikMichelson) [inline src/groups/groups.service.ts:44]:
```suggestion
        `A group with the name '${name}' already exists.`,
```

File: src/groups/groups.service.ts [hunk 8]
```diff
@@ -21,6 +21,35 @@ export class GroupsService {
...
+        'createGroup',
+      );
+      throw new AlreadyInDBError(
+        `A group with the username '${name}' already exists.`,
```
Comment (ErikMichelson) [inline src/groups/groups.service.ts:48]:
```suggestion
        `A group with the name '${name}' already exists.`,
```

**Review by ErikMichelson — CHANGES_REQUESTED** [review]

File: src/groups/groups.service.spec.ts [hunk 9]
```diff
@@ -46,6 +46,32 @@ describe('GroupsService', () => {
...
+        .mockImplementationOnce(async (group: Group): Promise<Group> => group);
+    });
+    it('works', async () => {
+      const user = await service.createGroup(groupName, displayname);
+      expect(user.name).toEqual(groupName);
+      expect(user.displayName).toEqual(displayname);
+    });
+    it('fails if username is already taken', async () => {
+      jest.spyOn(groupRepo, 'save').mockImplementationOnce(() => {
+        throw new Error();
```
Comment (davidmehren) [inline src/groups/groups.service.spec.ts:64]: This looks a bit suspicious. With this mock, `groupRepo.save` never succeeds, which should mean that `groupService.createGroup` also never succeeds.

**Review by davidmehren — CHANGES_REQUESTED** [review]

**Review by davidmehren — APPROVED** [review]

**Pull request merged**